src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -3534,8 +3547,11 @@ class runlist_impl
       return std::cv_status::no_timeout;
 
     // Wait throws on error. On timeout just return
-    if (wait(timeout) == std::cv_status::timeout)
+    if (wait(timeout) == std::cv_status::timeout) {
+      // Dump uC log buffer for debug when timeout happens
+      xrt_core::hw_context_int::dump_uc_log_buffer(m_hwctx);
```

Comment:
Isn't this dumping twice?  Appears to already be dumped in wait(...) up above.

Reply:
No Soren, this one is for runlist
The other one shown in above wait is in xrt::run_impl (for xrt::run)